latest release forgero-fabric-0.10.8+1.19.2 Startup crash on large modpack - likely mod interaction issue low priority

Startup crash on latest release.

initially presented as ftb chunks in first crashlog: https://pastebin.com/DRsxXNhi
removed ftb chunks, crashlog here: https://pastebin.com/ArX37Pa8

Caused by: java.lang.NullPointerException: Cannot invoke "c**om.sigmundgranaas.forgero.core.registry.StateCollection.all()" because "com.sigmundgranaas.forgero.core.ForgeroStateRegistry.STATES" is null**
	at com.sigmundgranaas.forgero.fabric.toolstats.ToolStatTagGenerator.generateTags(ToolStatTagGenerator.java:16)
	at com.sigmundgranaas.forgero.fabric.ForgeroCompatInitializer.onInitialize(ForgeroCompatInitializer.java:26)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)

This is in a large modpack, likely not an issue on your side.
Figured I would bring it to your attention before 12yo's in your curseforge comments.

Low priority. Thank you for your work. Cool mod :)

--- a/fabric/forgero-fabric-compat/src/main/java/com/sigmundgranaas/forgero/fabric/toolstats/ToolStatTagGenerator.java
+++ b/fabric/forgero-fabric-compat/src/main/java/com/sigmundgranaas/forgero/fabric/toolstats/ToolStatTagGenerator.java
@@ -12,6 +12,9 @@ import static com.sigmundgranaas.forgero.fabric.resources.ARRPGenerator.RESOURCE
 
 public class ToolStatTagGenerator {
 	public static void generateTags() {
+		if (ForgeroStateRegistry.STATES == null){
+			return;
+		}
 		var tag = new JTag();
 		ForgeroStateRegistry.STATES.all().stream()
 				.map(Supplier::get)

--- a/forgero-core/src/main/java/com/sigmundgranaas/forgero/core/ForgeroStateRegistry.java
+++ b/forgero-core/src/main/java/com/sigmundgranaas/forgero/core/ForgeroStateRegistry.java
@@ -11,12 +11,15 @@ import com.sigmundgranaas.forgero.core.state.State;
 import com.sigmundgranaas.forgero.core.state.StateProvider;
 import com.sigmundgranaas.forgero.core.type.TypeTree;
 
+import javax.annotation.Nullable;
+
 import java.util.*;
 import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 public class ForgeroStateRegistry {
+	@Nullable
 	public static StateCollection STATES;
 	public static List<StateProvider> CREATE_STATES;
 	public static Map<String, String> STATE_TO_CONTAINER;
@@ -30,7 +33,7 @@ public class ForgeroStateRegistry {
 	public static StateFinder stateFinder() {
 		return (String id) ->
 		{
-			if (isState().apply(id)) {
+			if (isState().apply(id) && STATES != null) {
 				return STATES.find(id).map(Supplier::get);
 			} else {
 				return Optional.empty();
@@ -39,7 +42,7 @@ public class ForgeroStateRegistry {
 	}
 
 	public static Function<String, Boolean> isState() {
-		return (id) -> STATES.contains(id) || CONTAINER_TO_STATE.containsKey(id);
+		return (id) -> (STATES != null && STATES.contains(id)) || CONTAINER_TO_STATE.containsKey(id);
 	}
 
 	public static ResourceListener<Map<String, State>> stateListener() {

